pipeline/misc/validate_mask_isohits.py

```diff
@@ -0,0 +1,200 @@
+        std_tt[tag] = np.std(cl00[tag], axis=0)[0]/norm_factor
+        std_ee[tag] = np.std(cl22[tag], axis=0)[0]/norm_factor
+        std_bb[tag] = np.std(cl22[tag], axis=0)[3]/norm_factor
+
+        std_bb_bonly[tag] = np.std(cl22b[tag], axis=0)[3]/norm_factor
+
+
+    ##########
+    # Plots
+    ##########
+    formatting = {'fmt': 'o', 'markersize': 0, 'linewidth': 2, 'capsize': 3, 'capthick': 2}
+    offsets = (np.arange(len(mask_tags)) - len(mask_tags)/2 + 0.5) * X_OFFSET_UNIT
+    ells = {tag: leff+offsets[i] for i, tag in enumerate(mask_tags)}
+    labels = {tag: tag[1:] if tag.startswith('_') else tag for tag in mask_tags}
+
+    plt.figure()
+    plt.semilogy(leff, cbtt, label='TT theory', color='k', linestyle='-')
+    for tag in mask_tags:
+        plt.errorbar(ells[tag], means_tt[tag], yerr=std_tt[tag], label=labels[tag], **formatting)
+    plt.legend()
+    plt.xlabel(r"Multipole $\ell$")
+    plt.ylabel(r"$C_b^{TT} \; [\mu\mathrm{K}^2]$")
+    plt.savefig(os.path.join(plot_dir, "mask_validation_TT.png"), bbox_inches='tight')
+    plt.close()
+
+    plt.figure()
+    plt.semilogy(leff, cbee, label='EE theory', color='k', linestyle='-')
+    for tag in mask_tags:
+        plt.errorbar(ells[tag], means_ee[tag], yerr=std_ee[tag], label=labels[tag], **formatting)
+    plt.legend()
+    plt.xlabel(r"Multipole $\ell$")
+    plt.ylabel(r"$C_b^{EE} \; [\mu\mathrm{K}^2]$")
+    plt.savefig(os.path.join(plot_dir, "mask_validation_EE.png"), bbox_inches='tight')
+    plt.close()
+
+    plt.figure()
+    plt.semilogy(leff, cbbb, label='BB theory', color='k', linestyle='-')
+    for tag in mask_tags:
+        plt.errorbar(ells[tag], means_bb[tag], yerr=std_bb[tag], label=labels[tag], **formatting)
+    plt.legend()
+    plt.xlabel(r"Multipole $\ell$")
+    plt.ylabel(r"$C_b^{BB} \; [\mu\mathrm{K}^2]$")
+    plt.savefig(os.path.join(plot_dir, "mask_validation_BB.png"), bbox_inches='tight')
+    plt.close()
+
+    plt.figure()
+    for tag in mask_tags:
+        plt.plot(ells[tag], (means_bb[tag]-cbbb)/std_bb[tag], label=labels[tag])
+    plt.xlabel(r"Multipole $\ell$")
+    plt.ylabel(r"$(\^{C}_b - C_b^{\mathrm{th}})/\sigma(\^{C}_b)$")
+    plt.savefig(os.path.join(plot_dir, "mask_validation_sigmas.png"), bbox_inches='tight')
+    plt.close()
+
+    plt.figure()
+    for tag in mask_tags:
+        plt.plot(ells[tag], std_bb[tag]/std_bb_bonly[tag], label=labels[tag])
+    plt.xlabel(r"Multipole $\ell$")
+    plt.ylabel(r"$\sigma(C_b^{\mathrm{CMB}})/\sigma(C_b^{\mathrm{B\ only}})$")
+    plt.savefig(os.path.join(plot_dir, "mask_validation_compare_bonly.png"), bbox_inches='tight')
+    plt.close()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--globals", required=True, help="Path to paramfile")
+    parser.add_argument("--fwhm", default=30, help="Beam size (arcmin). Default: 30")
+    parser.add_argument("--puretype", default='cmbTEB', help="Input sim pure type (T/E/B). Default: 'cmbTEB'")
```

Comment:
A bit peaky comment: can we not call this `puretype`, but maybe something like `simtype`? The reason is that `pure_type` is being used for the transfer function estimation sims, which are not CMB sims (containing B, EB, or TEB, etc.) but power-law Gaussian sims that carry only one type of polarization.